src/main/java/weizberg/citibike/StationMethods.java

```diff
@@ -0,0 +1,96 @@
+package weizberg.citibike;
+
+
+import io.reactivex.rxjava3.core.Single;
+import io.reactivex.rxjava3.schedulers.Schedulers;
+import weizberg.citibike.json.DataCollection;
+import weizberg.citibike.json.Station;
+
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.concurrent.TimeUnit;
+
+
+public class StationMethods {
+
+
+    private final Map<String, Station> stations;
+
+
+    public StationMethods() {
+        MergeStationData mergeStationData = new MergeStationData();
+        stations = mergeStationData.getStationsMap();
+    }
+
+    public int getNumDocks(String stationId) {
+        Station station = stations.get(stationId);
+        return station.num_docks_available;
+    }
+
+    public int getNumBikes(String stationId) {
+        Station station = stations.get(stationId);
+        return station.num_bikes_available;
+    }
+
+
+    public String closestPickUpStation(double lat, double lon) {
+        double closestDistance = 0;
+        String closestStationId = "";
+
+
+        for (int i = 0; i < stations.values().size(); i++) {
+            Station station = (Station) stations.values().toArray()[i];
```

Comment:
This is not a good way to loop through a map.

https://stackoverflow.com/a/1066607